custom-ops-introduction/README.md

```diff
@@ -0,0 +1,115 @@
+# Custom Operations: An Introduction to Programming GPUs and CPUs with Mojo
+
+In this recipe, we will cover:
+
+* How to extend a MAX Graph using custom operations.
+* Using Mojo to write high-performance calculations that run on GPUs and CPUs.
+* The basics of GPU programming in MAX.
+
+We'll walk through running three examples that show
+
+* adding one to every number in an input tensor
+* performing hardware-specific addition of two vectors
+* and calculating the Mandelbrot set on CPU and GPU.
+
+Let's get started.
+
+## Requirements
+
+Please make sure your system meets our
+[system requirements](https://docs.modular.com/max/get-started).
+
+To proceed, ensure you have the `magic` CLI installed:
+
+```bash
+curl -ssL https://magic.modular.com/ | bash
+```
+
+or update it via:
+
+```bash
+magic self-update
+```
+
+### GPU requirements
+
+These examples can all be run on either a CPU or GPU. To run them on a GPU,
+ensure your system meets
+[these GPU requirements](https://docs.modular.com/max/faq/#gpu-requirements):
+
+* Officially supported GPUs: NVIDIA Ampere A-series (A100/A10), or Ada
+  L4-series (L4/L40) data center GPUs. Unofficially, RTX 30XX and 40XX series
+  GPUs have been reported to work well with MAX.
+* NVIDIA GPU driver version 555 or higher. [Installation guide here](https://www.nvidia.com/download/index.aspx).
+
+## Quick start
+
+1. Download the code for this recipe using git:
+
+```bash
+git clone https://github.com/modular/max-recipes.git
+cd max-recipes/custom-ops-introduction
+```
+
+2. Run each of the examples:
+
+```bash
+magic run add_one
+magic run vector_addition
+magic run mandelbrot
+```
+
+3. Browse through the commented source code to see how they work.
+
+## Custom operation examples
```

Comment:
Do you mind adding more explanations? I know we're targeting GTC and the docs position but something that we can later also port to docs later if needed.

Reply:
I've expanded these significantly, providing a lot more detail about what's going on here.